src/mgxs_interface.cpp

```diff
@@ -19,41 +19,52 @@
 namespace openmc {
 
 //==============================================================================
-// Global variable definitions
+// Mgxs data loading interface methods
 //==============================================================================
 
 namespace data {
+  MgxsInterface mgInterface;
+}
 
-int num_energy_groups;
-int num_delayed_groups;
-std::vector<double> energy_bins;
-std::vector<double> energy_bin_avg;
-std::vector<double> rev_energy_bins;
-
-} // namesapce data
+MgxsInterface::MgxsInterface(const std::string& path_cross_sections,
+                             const std::vector<std::string> xs_to_read,
+                             const std::vector<std::vector<double>> xs_temps)
+{
+  read_header(path_cross_sections);
+  set_nuclides_and_temperatures(xs_to_read, xs_temps);
+  init();
```

Comment:
I generally like to see `this->` when one member function is called from another just to make clear that it's not calling some other function, but I'll leave it up to you if you want to change it.

Reply:
Hm, OK, not changed in that case.